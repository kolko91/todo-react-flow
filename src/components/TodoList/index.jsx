import React from 'react';
import styled from 'styled-components';

import Todo from 'components/Todo';

import type { TodosState, Id } from 'types/todos';

const List = styled.ul`
    margin: 0;
    padding: 0;
    list-style: none;
`;

export type Props = {
  todos: TodosState,
  onTodoClick: (id: Id) => void
};

const TodoList = ({ todos, onTodoClick }: Props) => (
  <List>
    {todos.data.map(todo => (
      <Todo key={todo.id} {...todo} onClick={() => onTodoClick(todo.id)} />
    ))}
  </List>
);

export default TodoList;
