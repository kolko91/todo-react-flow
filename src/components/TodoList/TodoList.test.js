import React from 'react';
import renderer from 'react-test-renderer';
import TodoList from './index';


describe('component', () => {
  describe('TodoList', () => {
    it('should render correctly', () => {
      const props = {
        onSave: jest.fn(),
        todos: {
          data: [
            {
              id: 1,
              text: 'title1',
              completed: true,
            },
            {
              text: 'test2',
              completed: false,
              id: 4,
            },
          ],
          loaded: false,
          error: false,
        },
      };
      const tree = renderer.create(<TodoList {...props} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
