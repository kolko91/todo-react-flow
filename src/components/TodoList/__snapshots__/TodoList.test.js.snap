// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component TodoList should render correctly 1`] = `
<ul
  className="sc-EHOje iRXTJc"
>
  <li
    className="sc-ifAKCX hdYSEK"
  >
    <div>
      <input
        checked={true}
        className="sc-bwzfXH dydaOf"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="sc-htpNat dDFmjJ"
        onDoubleClick={[Function]}
      >
        title1
      </label>
      <button
        className="sc-bxivhb bKUsik"
        onClick={[Function]}
      />
    </div>
  </li>
  <li
    className="sc-ifAKCX kYTDwW"
  >
    <div>
      <input
        checked={false}
        className="sc-bwzfXH dydaOf"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="sc-htpNat dDFmjJ"
        onDoubleClick={[Function]}
      >
        test2
      </label>
      <button
        className="sc-bxivhb bKUsik"
        onClick={[Function]}
      />
    </div>
  </li>
</ul>
`;
